clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java

```diff
@@ -29,13 +30,13 @@
 
 /**
  * Provides a stream API to read a block from Tachyon. An instance extending this class can be
- * obtained by calling {@link TachyonBlockStore#getInStream(long)}. The buffer size of the stream
- * can be set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a
- * block. This class is not thread safe and should only be used by one thread.
+ * obtained by calling {@link TachyonBlockStore#getInStream}. The buffer size of the stream can be
+ * set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a block.
  *
  * This class provides the same methods as a Java {@link InputStream} with additional methods from
  * Tachyon Stream interfaces.
  */
+@NotThreadSafe
```

Comment:
shall we use `@NotThreadSafe` explicitly? if not, shall we mark all non-thread-safe classes with this annotation?
Alternatively, can we assume classes without `@ThreadSafe` is unsafe?